### PR TITLE
Initial dota bot

### PR DESCRIPTION
**Pull request by tianhongtan**
Created the dal and several commands executing against the opendota api.

Example bot executions:
https://imgur.com/a/V63Wotu

### CODE REVIEW

File: commands/dota.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+function formatSeconds(inputSeconds) {
+  const minutes = Math.floor(inputSeconds / 60)
+  const seconds = inputSeconds % 60
+  return `${minutes} ${minutes == 1 ? 'minute' : 'minutes'} and ${seconds} ${seconds == 1 ? 'second' : 'seconds'}`
+}
+
+/**
+ * Generates the dotaConf composite key from channelId, discordId, and dotaID
+ */
+function getConfKey(channelId, discordId, dotaId) {
```
Comment (mynameisivan) [inline commands/dota.js:98]: shouldn't dota id be left out of the key? 

if a user wants to bind multiple ids to themselves, maybe we can store a list of ids? 
(discord channel + discord user id) => [ list, of, dota, ids] 

or maybe we just support 1 dota user acc per discord user (per server)

Comment (tianhongtan) [reply commands/dota.js:98]: That does work better, let me update.

File: commands/dota.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+function sendHelp(channel) {
+  return channel.send(`= USAGE =
+.dota register [dota account id]  :: Links your discordId to the specified dota account id for the current channel. Account id can be found in your dotabuff profile url (https://www.dotabuff.com/players/<ACCOUNT_ID>)
+.dota getLatestMatch :: gets the latest match for your linked dota account(s).`,
+  {code: 'asciidoc'});
+}
+
+function getDiscordUsername(client, discordId) {
```
Comment (I-AM-STEVEe) [inline commands/dota.js:113]: We should make this a util method, I had to do the same thing in the other command

Comment (tianhongtan) [reply commands/dota.js:113]: I've added this to functions.js as part of the client
usage (tested in debugger): 
```
>> client.getDiscordUsername(discordUser.id)
>>'trollpolice'
```

**Pull request merged**